auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java

```diff
@@ -0,0 +1,41 @@
+package org.aerogear.mobile.security;
+
+import android.content.Context;
+
+import org.aerogear.mobile.core.metrics.MetricsService;
+import org.aerogear.mobile.security.metrics.SecurityCheckResultMetric;
+
+import java.util.Collection;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class AsyncSecurityCheckExecutor extends AbstractSecurityCheckExecutor {
+
+    private final ExecutorService pool = Executors.newFixedThreadPool(10);
+
+    AsyncSecurityCheckExecutor(final Context context, final Collection<SecurityCheck> checks, final MetricsService metricsService) {
+        super(context, checks, metricsService);
+    }
+
+    public Future<SecurityCheckResult>[] execute() {
```

Comment:
can we change the interface ?

Reply:
@wei-lee @aidenkeating What do you think? I'm ok with changing the interface to return a `Collection` instead of an Array